### DISCUSSION TITLE
Need help to understand erroneous re-renders in a simple form usage

### DISCUSSION THREAD

**Discussion by repl-ullas-ml** [react-hook-form/react-hook-form#6981] [Q&A]
In the codesandbox example https://codesandbox.io/s/react-hook-form-form-context-forked-rurkt , a `Dialog` is wrapped within a simple `FormProvider`.

I fail to understand how a simple `console.log("Test with/without this", formState.errors);` increases the render count. Also do try without the console log.

Please note: Our production app is experiencing erroneous re-renders and in the effort to debug/address, a simple minimalistic example has been provided here

Could anyone explain whats happening here? Is it a potential bug ?

![mix-match-collectors](https://user-images.githubusercontent.com/11545334/140430014-05d494a2-76ca-4c48-9fed-8350a2562675.gif)

**Comment by bluebill1049** [comment]
your onChange is firing validation for each keystroke.

```
const onChange = useCallback(
(event) => {
  setValue("note", event.target.value, {
    shouldDirty: true,
    shouldValidate: true
  });
},
[setValue]
);
```

**Reply by repl-ullas-ml** [reply]
Oh ok. But a use case we have is the user could enter only certain number of characters and once if he exceeds we have a yup resolver in place to raise an error and which in turn is propagated to the UI.

So is there not a way to deal with use case and not re-render ? Is the  `shouldValidate ` on every keystroke the only necessary evil to support this use case

**Reply by repl-ullas-ml** [reply]
@bluebill1049 What are the possible ways to circumvent the above issue? To better provide context i have updated the code sandbox
https://codesandbox.io/s/react-hook-form-form-context-forked-luvxi

Manually invoking `setError` only for the that particular field also re-renders the entire dialog. 

``` <FormBody setManualError={true} /> ```
